### PR TITLE
[MLIR][DISC] pattern conversion from tf2mhlo: ConvertSplitOpDynamic, ConvertSliceOpDynamic

### PR DESCRIPTION
**Pull request by azazhu**
We are porting our MLIR-based dynamic shape compiler to tf community (From OP def, Patttern, to Optimization pass, etc).
This is the third PR about tf2mhlo pattern conversion, which including ConvertSplitOpDynamic and ConvertSliceOpDynamic.
The rest pattern conversions we will add:
- ConvertSqueezeOpxxx
- ConvertStridedSliceOpxxx
- ConvertTileOpxxx
- ConvertTopKV2Opxxx
- ConvertUnpackOpxxx
- ConvertPrintOp
- ConvertSigmoidGradOpxxx
- ConvertSignOpxxx

### CODE REVIEW

**Comment by joker-eph** [comment]
Can you add test coverage for this?

**Comment by azazhu** [comment]
> Can you add test coverage for this?

added

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 1]
```diff
@@ -2900,6 +2900,75 @@ class ConvertSoftmaxOp : public OpRewritePattern<OpTy> {
...
+class ConvertSliceOpDynamic : public OpRewritePattern<TF::SliceOp> {
+ public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TF::SliceOp op,
+                                     PatternRewriter& rewriter) const override {
+    auto loc = op.getLoc();
+    auto input = op.input();
+    auto begin_indices = op.begin();
+    auto sizes = op.size();
```
[the remark this answers is not in the document]
Comment (joker-eph) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:3392]: (sorry I wrote this 3 weeks ago but didn't send it properly...)

[the remark this answers is not in the document]
Comment (azazhu) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc]: :)

[the remark this answers is not in the document]
Comment (azazhu) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc]: Done. BTW. If one commit fail on CI (such as "AMD ROCm -- Community CI Build"). The details link is unable to open usually. Is there any alternate method to get the failure information?

**Review by joker-eph — APPROVED** [review]

**Review by joker-eph — APPROVED** [review]

**Comment by gbaned** [comment]
@azazhu  Can you please resolve conflicts? Thanks!

**Comment by azazhu** [comment]
> @azazhu Can you please resolve conflicts? Thanks!

Done

**Review by joker-eph — APPROVED** [review]

**Pull request merged**